spec/acceptance/pe_metric_spec.rb

```diff
@@ -8,21 +8,90 @@
+    it { expect(service('puppet_ace-tidy.timer')).to be_running }
+    it { expect(service('puppet_bolt-metrics.timer')).to be_running }
+    it { expect(service('puppet_bolt-tidy.timer')).to be_running }
+    it { expect(service('puppet_orchestrator-metrics.timer')).to be_running }
+    it { expect(service('puppet_orchestrator-tidy.timer')).to be_running }
+    it { expect(service('puppet_postgres-metrics.timer')).to be_running }
+    it { expect(service('puppet_postgres-tidy.timer')).to be_running }
+    it { expect(service('puppet_puppetdb-metrics.timer')).to be_running }
+    it { expect(service('puppet_puppetdb-tidy.timer')).to be_running }
+    it { expect(service('puppet_puppetserver-metrics.timer')).to be_running }
+    it { expect(service('puppet_puppetserver-tidy.timer')).to be_running }
+    it { expect(service('puppet_system_cpu-metrics.timer')).to be_running }
+    it { expect(service('puppet_system_cpu-tidy.timer')).to be_running }
+    it { expect(service('puppet_system_memory-metrics.timer')).to be_running }
+    it { expect(service('puppet_system_memory-tidy.timer')).to be_running }
+    it { expect(service('puppet_system_processes-metrics.timer')).to be_running }
+    it { expect(service('puppet_system_processes-tidy.timer')).to be_running }
+  end
+  it 'creates tidy services files' do
     files = run_shell('ls /etc/systemd/system/puppet_*-tidy.service').stdout
     expect(files.split("\n").count).to eq(9)
   end
-  it 'check for the timer files' do
+  it 'creates the timer files' do
     files = run_shell('ls /etc/systemd/system/puppet_*-tidy.timer').stdout
     expect(files.split("\n").count).to eq(9)
   end
-  it 'checks if sysstat package is installed' do
+  it 'sysstat package is installed' do
     expect(package('sysstat')).to be_installed
   end
+
+  describe file('/opt/puppetlabs/puppet-metrics-collector/puppetserver/127.0.0.1') do
+    before(:each) { run_shell('systemctl start puppet_puppetserver-metrics.service') }
+
+    it { is_expected.to be_directory }
+    it 'contains metric files' do
+      files = run_shell('ls /opt/puppetlabs/puppet-metrics-collector/puppetserver/127.0.0.1/').stdout
+      expect(files.split('\n')).not_to be_empty
+    end
+  end
+
+  describe file('/opt/puppetlabs/puppet-metrics-collector/puppetdb/127.0.0.1') do
+    before(:each) { run_shell('systemctl start puppet_puppetdb-metrics.service') }
+
+    it { is_expected.to be_directory }
+    it 'contains metric files' do
+      files = run_shell('ls /opt/puppetlabs/puppet-metrics-collector/puppetdb/127.0.0.1/').stdout
+      expect(files.split('\n')).not_to be_empty
+    end
+  end
+
+  describe file('/opt/puppetlabs/puppet-metrics-collector/orchestrator/127.0.0.1') do
+    before(:each) { run_shell('systemctl start puppet_orchestrator-metrics.service') }
+
+    it { is_expected.to be_directory }
+    it 'contains metric files' do
+      files = run_shell('ls /opt/puppetlabs/puppet-metrics-collector/orchestrator/127.0.0.1/').stdout
```

Comment:
I think taking advantage of shell globbing would be better than testing if a newline appears in `ls` output.  The default is for `nullglob` to be turned off, meaning a glob pattern that doesn't match expands to itself.  e.g.

```
$ ls /tmp/foo/*
ls: cannot access /tmp/foo/*: No such file or directory
```

Can you try this loop and let me know if it works?

```
for f in /opt/puppetlabs/puppet-metrics-collector/puppetserver/127.0.0.1/*; do [[ -e $f ]] && exit 0; done
```

The exit code should be `0` if it finds files and `1` if it does not.

Reply:
Sounds good. I switched to using the `nullglob`. The `run_shell` will fail the spec test if it has a nonzero exit code. If it comes back without an exit code and no output, the expect will catch the failure.